### PR TITLE
dpms: Add format

### PR DESCRIPTION
**Pull request by lasers**
This adds format: string to display (default '~~{query}~~{dpms}'). Closes https://github.com/ultrabug/py3status/issues/617. 

Ongoing efforts to `FORMAT ALL THE THINGS!` (meme). https://github.com/ultrabug/py3status/issues/98

### CODE REVIEW

File: py3status/modules/dpms.py [hunk 1]
```diff
@@ -7,12 +7,16 @@
...
 
 Configuration parameters:
-    format_off: string to display when DPMS is disabled (default 'DPMS')
-    format_on: string to display when DPMS is enabled (default 'DPMS')
+    format: string to display (default '{query}')
+    format_off: string to display when disabled (default 'DPMS')
+    format_on: string to display when enabled (default 'DPMS')
+
+Format placeholders:
+    {query} display current dpms setting
```
Comment (tobes) [inline py3status/modules/dpms.py]: `{icon}` might be better see last comment on https://github.com/ultrabug/py3status/issues/617

File: py3status/modules/dpms.py [hunk 2]
```diff
@@ -7,12 +7,16 @@
...
 
 Color options:
-    color_on: when dpms is enabled, defaults to color_good
-    color_off: when dpms is disabled, defaults to color_bad
+    color_on: when enabled, defaults to color_good
+    color_off: when disabled, defaults to color_bad
```
Comment (tobes) [inline py3status/modules/dpms.py]: I prefer the explicit comment

File: py3status/modules/dpms.py (hunk 1 above)
Comment (lasers) [inline py3status/modules/dpms.py]: If DPMS is enabled: Can use `format_on=`: `DPMS`, `DPMS: ON`, `DPMS+`, `● DPMS`, `●`, etc.

If DPMS is disabled: Can use `format_off=`: `DPMS`, `DPMS: OFF`, `DPMS-`, `○ DPMS`, `○`, etc.

File: py3status/modules/dpms.py (hunk 2 above)
Comment (lasers) [inline py3status/modules/dpms.py]: Fixed.

File: py3status/modules/dpms.py (hunk 1 above)
Comment (lasers) [inline py3status/modules/dpms.py]: I looked it again. `{dpms}` ?

**Comment by tobes** [comment]
`{format_dpms}` is bad because this is a placeholder not a format and also the name is long.

I still prefer `{icon}` or `{state}` so it'd be like this https://github.com/ultrabug/py3status/blob/master/py3status/modules/README.md#file_status

**Comment by lasers** [comment]
Looking at other modules... What about just `{dpms}`?
`{icon}` seems wrong because one could do this.
If not, `{state}` seems acceptable. Let me know.

```
format = "dpms: {icon}"
format_on = "enabled"
format_on = "disabled"
```

```
format = "dpms: {state}"
format_on = "enabled"
format_on = "disabled"
```

```
format = "dpms: {dpms}"
format_on = "enabled"
format_on = "disabled"
```

**Comment by tobes** [comment]
The thing to bear in mind is that the modules have been created by different users at different times.  The project is now at a point where we need to start making things so that they behave in similar ways so that users can quickly configure them etc, allow functionality to be shared across all modules.  we also need to respect users and not break there setups.

My opinion and this is just my opinion is that we should have standard config namings

`format_...` these describe the formats are processed by the formatter so have all that functionality.
Ideally modules should have a single `format` but in some cases this is not possible

`icon_...` used to allow user icons/text used for placeholders

`button_...` used to select what button performs an action

this is a long process and sometimes we need to make changes to core functionality to allow modules to be updated, eg we can now allow configuration options to be added/removed/renamed using the deprecation stuff.  For some modules/users needs we have have had to add new formatter options for example.  Sometimes changes to modules will happen over a period of time rather than all at once.  But when naming things (probably the hardest aspect of coding by a long way)  it is good to think things through as undoing these changes is painful.

**Comment by tobes** [comment]
My preference is still to have a config like this
```
dpms {
    format = "dpms: {icon}"
    icon_on = "enabled"
    icon_off = "disabled"
}
```

We have removed the old `format_on` etc.  To prevent breaking user configs we can make use of the deprecation stuff https://github.com/ultrabug/py3status/tree/master/doc#deprecation

File: py3status/modules/dpms.py [hunk 3]
```diff
@@ -7,8 +7,12 @@
...
 
 Configuration parameters:
-    format_off: string to display when DPMS is disabled (default 'DPMS')
-    format_on: string to display when DPMS is enabled (default 'DPMS')
+    format: string to display (default '{state}')
+    icon_off: string to display when dpms is disabled (default 'DPMS')
+    icon_on: string to display when dpms is enabled (default 'DPMS')
+
+Format placeholders:
+    {state} display current dpms state
```
Comment (tobes) [inline py3status/modules/dpms.py]: `{icon}` makes sense if the params are `icon_on` etc

File: py3status/modules/dpms.py [hunk 4]
```diff
@@ -24,20 +28,35 @@ class Py3status:
...
+    icon_off = "DPMS"
+    icon_on = "DPMS"
+
+    class Meta:
+        deprecated = {
+            'rename': [
+                {
+                    'param': 'format_on',
+                    'new': 'icon_on',
+                    'msg': 'obsolete parameter // use `icon_on`',
```
Comment (tobes) [inline py3status/modules/dpms.py]: I'd remove the `//` as it has no meaning and adds confusion.

Comment (lasers) [inline py3status/modules/dpms.py]: It helps separate things in one-liner (in my log). Will remove.

**Comment by ultrabug** [comment]
thanks a lot for this ongoing effort !

**Pull request merged**